m4 build is failing

```
gcc -std=gnu99  -I.     -g -O2 -MT clean-temp.o -MD -MP -MF .deps/clean-temp.Tpo -c -o clean-temp.o clean-temp.c
In file included from clean-temp.h:22:0,
                 from clean-temp.c:23:
./stdio.h:477:1: error: 'gets' undeclared here (not in a function)
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 ^
make[3]: *** [clean-temp.o] Error 1
```

$ gcc --version
gcc (Debian 4.8.2-15) 4.8.2

--- a/pkgs/m4.yaml
+++ b/pkgs/m4.yaml
@@ -3,8 +3,8 @@ dependencies:
   build: []
 
 sources:
-  - url: http://ftp.gnu.org/gnu/m4/m4-1.4.16.tar.bz2
-    key: tar.bz2:ijiayh653ef4wfq6nnmn5ey5o2mstrev
+  - url: http://ftp.gnu.org/gnu/m4/m4-1.4.17.tar.bz2
+    key: tar.bz2:rzhb7frzgijw5vc5zvnpwddoen7jnjx4
 
 when_build_dependency:
   - prepend_path: PATH

